File: Codeforces/Round 272/C.cpp
#include <bits/stdc++.h>
using namespace std;
#define f(i,n) for(int i=0;i<n;i++)
#define fab(i,a,b) for(int i=a;i<=b;i++)
#define g(n) scanf("%lld",&n)
const long long int modulo = 1000000007;
typedef long long int lli;
const lli inv = 500000004;

lli partfunc(lli a, lli b, lli r)
{
	lli armod = (a*r)%modulo;
	lli ans = ((b*(a+1))/2 + 1)%modulo;
	ans = (ans*armod)%modulo;
	return ans;
}

int main()
{
	lli a,b;
	g(a),g(b);
	lli s = 0;
	for(lli r = 1; r<=b-1; r++)
	{
		s += partfunc(a,b,r);
		s%=modulo;
	}
	printf("%lld\n",s);
	return 0;
}

File: Codeforces/Round 272/C.py
modulo = 1000000007;
inv = 500000004;
def partfunc(a,b):
	s = (a*b*(a+1))/2 + a
	s *= ((b-1)*(b))/2
	s %= modulo
	return s

def check(a,b):
	s = 0
	x = 1
	m = (b-1)*(a*b+1)
	while x<=m:
		q = x/b
		r = x%b
		if r>0:
			if q%r==0 and q/r <= a and q/r >= 1:
				s += x
		x += 1
	return s

def main():
	r = raw_input().split(' ')
	a = int(r[0])
	b = int(r[1])
	print partfunc(a,b)
	# print check(a,b)

main()